src/main/java/com/mikahil/tarasevich/dao/impl/CourseDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+package com.mikahil.tarasevich.dao.impl;
+
+import com.mikahil.tarasevich.dao.ConnectorDB;
+import com.mikahil.tarasevich.dao.CourseDao;
+import com.mikahil.tarasevich.dao.exception.DataBaseSqlRuntimeException;
+import com.mikahil.tarasevich.entity.Course;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Optional;
+
+public class CourseDaoImpl extends AbstractPageableCrudDaoImpl<Course> implements CourseDao {
+
+    private static final String SAVE_COURSE_QUERY = "INSERT INTO school.courses (course_name, description) VALUES(?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM school.courses WHERE id=?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM school.courses";
+    private static final String FIND_BY_COURSE_NAME_QUERY = "SELECT * FROM school.courses WHERE course_name=?";
+    private static final String FIND_COURSES_BY_STUDENT_ID_QUERY =
+            "SELECT courses.id, course_name, courses.description\n" +
+                    "FROM school.student_courses\n" +
+                    "LEFT JOIN school.students ON students.id = student_id\n" +
+                    "LEFT JOIN school.courses ON courses.id = course_id\n" +
+                    "WHERE students.id = ?";
+    private static final String UPDATE_COURSE_QUERY = "UPDATE school.courses SET course_name = ?," +
+            "description = ? WHERE id= ?";
+    private static final String DELETE_COURSE_QUERY = "DELETE FROM school.courses WHERE id=?";
+    private static final String DELETE_COURSE_DEPENDENCIES_QUERY =
+            "DELETE FROM school.student_courses WHERE course_id=?";
+
+    public CourseDaoImpl(ConnectorDB connector) {
+        super(connector, FIND_BY_ID_QUERY, FIND_ALL_QUERY, DELETE_COURSE_QUERY);
+    }
+
+    @Override
+    public Course save(Course course) {
+        try (final PreparedStatement preparedStatement = connector
+                .getConnection().prepareStatement(SAVE_COURSE_QUERY, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, course.getCourseName());
+            preparedStatement.setString(2, course.getDescription());
+            int courseId = getGeneratedId(preparedStatement);
+            return makeCourseWithId(course, courseId);
+        } catch (SQLException e) {
+            //log
+            throw new DataBaseSqlRuntimeException("", e);
+        }
+    }
+
+    @Override
+    public void saveAll(List<Course> courses) {
+        try (final PreparedStatement preparedStatement = connector
```

Comment:
move impelentation to AbstractCrudDao Impl, same about all crud methods in all calsses